test/common/index.js

```diff
@@ -759,3 +759,19 @@ exports.getTTYfd = function getTTYfd() {
   }
   return tty_fd;
 };
+
+// Hijack stdout and stderr
+function hijackStdWritable(name, listener) {
+  const stream = process[name];
+  const _write = stream.write.bind(stream);
+
+  stream.writeTimes = 0;
+  stream.write = function(data) {
+    listener(data);
+    _write(data);
+    stream.writeTimes++;
```

Comment:
This logic seem not go well with the [writable.write](https://nodejs.org/dist/latest-v8.x/docs/api/stream.html#stream_writable_write_chunk_encoding_callback) specification: 
> callback (Function) :Callback for when this chunk of data is flushed

Your logic will cause:
- Issue the completion handler before the actual write (_write)
- Listener will be called in the same loop tick, making it a direct call, not a callback.

I propose:
`_wite(data, listener)`